src/k2/persistence/plog_client/PlogClient.h

```diff
@@ -47,24 +47,27 @@ class PlogClient {
     ~PlogClient();
 
     // obtain the persistence cluster for a given name from the cpo and obtain the endpoints of the plog server
-    seastar::future<> init(String clusterName);
+    seastar::future<> init(String clusterName, String cpo_url);
 
     // allow users to select a specific persistence group by its name
     bool selectPersistenceGroup(String name);
 
     // create a plog with retry times
-    // TODO: revise this method, making this retry as an internal config variable instead of parameter.
+    // TODO: revise this method, making this retry as an internal config variable instead of parameter. 
     seastar::future<std::tuple<Status, String>> create(uint8_t retries = 1);
 
-    // append a payload into a plog at the given offset
+    // append a payload into a plog at the given offset 
     seastar::future<std::tuple<Status, uint32_t>> append(String plogId, uint32_t offset, Payload payload);
 
-    // read a payload
+    // read a payload 

```

Comment:
we aren't reading a `payload`. I think a better comment would be:
// read the given `size` bytes from the PLOG with the given `plogId`, starting to read at the given `offset`
// The result is returned as a single Payload containing the requested number of bytes
// what happens if there are not enough bytes available?
// pagination/continuation?

Reply:
changed

Reply:
instead of return an error when the size is large, we will return the current data we could return

Reply:
fixed